src/Stores/task.ts

```diff
@@ -1,13 +1,16 @@
+import { ref } from "vue";
 import { defineStore } from "pinia";
 import type { Task } from "@/type/home";
 
-export const useTaskStore = defineStore("task", {
-  state: () => ({
-    taskToEdit: null as Task | null,
-  }),
-  actions: {
-    setTaskToEdit(task: Task) {
-      this.taskToEdit = task;
-    },
-  },
-});
\ No newline at end of file
+export const useTaskStore = defineStore("task", () => {
+  const taskToEdit = ref<Task | null>(null);
+
+  function setTaskToEdit(task: Task) {
+    taskToEdit.value = task;
+  }
```

Comment:
Do we need this function? Since the function is only changing the `taskToEdit`. We can change the `taskToEdit` value directly without using this function.